lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -669,7 +669,8 @@ struct ConvertLayoutOpConversion
       // TODO: Support types other than float16 and
       // bf16 (represented as int16 in llvm ir).
       assert((type::isFloat(elemTy) || type::isInt(elemTy)) && elemSize == 16);
-      unsigned vecSize = 4;
+      auto dotOperandLayout = dstTy.getEncoding().cast<triton::gpu::DotOperandEncodingAttr>();
+      unsigned vecSize = dotOperandLayout.getKWidth() / 2;
```

Comment:
From this line `unsigned kWidth = 32 / bitwidth`; `kWidth = 2` for fp16/bf16, then `vecSize = 1`, do I misunderstand something?

Reply:
nv and amd gpus are both using the `kWidth` to indicate the k dim. What you saw is probably how it is computed for nv gpus (A100). 
I think Alek hasn't updated the kWidth computation yet.

@binarman  For vecSize, since you set it to `kWidth`/2, I assume you mean kWidth = mfmaInstM*mfmaInstK/64?

Reply:
@scxiao @zhanglx13 

> For vecSize, since you set it to kWidth/2, I assume you mean kWidth = mfmaInstM*mfmaInstK/64?

Yes, this `vecSize` is a number of elements per one mfma instruction per one lane (thread).

MFMA 32x32 A and B operands are distributed between 64 lanes like this (fp32/bfloat16 32x32x8 case):
![Screenshot from 2023-08-19 18-41-25](https://github.com/ROCmSoftwarePlatform/triton/assets/3631351/c224c4c0-a17f-4c52-91cb-9d81e16b8b8a)

kWidth holds total K width of operand per instruction.

Reply:
I just noticed that `kWidth` means the number of elements per thread along the k dim, rather than the k dim.  Can we redesign our usage of `kWidth` so that we (amd nd nv) refer to the same thing?

Reply:
here is the PR: https://github.com/ROCmSoftwarePlatform/triton/pull/326